src/core/reducers/abuse.js

```diff
@@ -124,8 +129,42 @@ export function showAddonAbuseReportUI({
   };
 }
 
+type InitiateAddonAbuseReportViaFirefoxParams = {|
+  _reportAbuse: typeof reportAbuse,
+  addon: AddonType,
+|};
+
+export type InitiateAddonAbuseReportViaFirefoxAction = {|
+  type: typeof INITIATE_ADDON_ABUSE_REPORT_VIA_FIREFOX,
+  payload: InitiateAddonAbuseReportViaFirefoxParams,
+|};
+
+export function initiateAddonAbuseReportViaFirefox({
+  _reportAbuse = reportAbuse,
```

Comment:
Why would we need to pass this util object here? We usually don't pass the `api` objects in the reducer.

Reply:
In a way their intents are identical, but the original one is named in a wau I wasn't able to get the mocking in the test to work the same way we do with `mockAPI`. I believe this is because we're not mocking an object with multiple functions, but rather this is a single function. If there's a different way to get it working in the test without having to pass the mock in explicitly, I'm happy to do that, but I couldn't figure one out, nor could I find any documentation or posts about how to do this with a single function.

Reply:
How about this?

```diff
diff --git a/src/core/reducers/abuse.js b/src/core/reducers/abuse.js
index 0d6a4f163..1a58a2439 100644
--- a/src/core/reducers/abuse.js
+++ b/src/core/reducers/abuse.js
@@ -1,7 +1,6 @@
 /* @flow */
 import invariant from 'invariant';
 
-import { reportAbuse } from 'core/addonManager';
 import type { AddonType } from 'core/types/addons';
 import type { AbuseReporter } from 'core/api/abuse';
 
@@ -130,7 +129,6 @@ export function showAddonAbuseReportUI({
 }
 
 type InitiateAddonAbuseReportViaFirefoxParams = {|
-  _reportAbuse: typeof reportAbuse,
   addon: AddonType,
 |};
 
@@ -140,14 +138,13 @@ export type InitiateAddonAbuseReportViaFirefoxAction = {|
 |};
 
 export function initiateAddonAbuseReportViaFirefox({
-  _reportAbuse = reportAbuse,
   addon,
 }: InitiateAddonAbuseReportViaFirefoxParams): InitiateAddonAbuseReportViaFirefoxAction {
   invariant(addon, 'addon is required');
 
   return {
     type: INITIATE_ADDON_ABUSE_REPORT_VIA_FIREFOX,
-    payload: { _reportAbuse, addon },
+    payload: { addon },
   };
 }
 
diff --git a/src/core/sagas/abuse.js b/src/core/sagas/abuse.js
index 7cc02db84..6ef402a84 100644
--- a/src/core/sagas/abuse.js
+++ b/src/core/sagas/abuse.js
@@ -10,6 +10,7 @@ import {
   loadAddonAbuseReport,
 } from 'core/reducers/abuse';
 import { createErrorHandler, getState } from 'core/sagas/utils';
+import { reportAbuse } from 'core/addonManager';
 import type { ReportAddonParams } from 'core/api/abuse';
 import type {
   InitiateAddonAbuseReportViaFirefoxAction,
@@ -48,9 +49,9 @@ export function* reportAddon({
 }
 
 export function* reportAddonViaFirefox({
-  payload: { _reportAbuse, addon },
+  payload: { addon },
 }: InitiateAddonAbuseReportViaFirefoxAction): Saga {
-  const abuseReported = yield _reportAbuse(addon.guid);
+  const abuseReported = yield reportAbuse(addon.guid);
   if (abuseReported) {
     yield put(
       loadAddonAbuseReport({
diff --git a/tests/unit/core/sagas/test_abuse.js b/tests/unit/core/sagas/test_abuse.js
index 971181d29..14bc5d7e9 100644
--- a/tests/unit/core/sagas/test_abuse.js
+++ b/tests/unit/core/sagas/test_abuse.js
@@ -10,6 +10,7 @@ import abuseReducer, {
 } from 'core/reducers/abuse';
 import apiReducer from 'core/reducers/api';
 import abuseSaga from 'core/sagas/abuse';
+import * as addonManager from 'core/addonManager';
 import {
   createFakeAddonAbuseReport,
   createStubErrorHandler,
@@ -20,11 +21,13 @@ import {
 describe(__filename, () => {
   let errorHandler;
   let mockApi;
+  let mockAddonMananger;
   let sagaTester;
 
   beforeEach(() => {
     errorHandler = createStubErrorHandler();
     mockApi = sinon.mock(api);
+    mockAddonMananger = sinon.mock(addonManager);
     const initialState = dispatchSignInActions().state;
     sagaTester = new SagaTester({
       initialState,
@@ -117,7 +120,6 @@ describe(__filename, () => {
     function _initiateAddonAbuseReportViaFirefox(params) {
       sagaTester.dispatch(
         initiateAddonAbuseReportViaFirefox({
-          _reportAbuse: sinon.stub(),
           addon: fakeAddon,
           ...params,
         }),
@@ -127,20 +129,23 @@ describe(__filename, () => {
     it('calls reportAbuse Firefox API', async () => {
       const addon = { ...fakeAddon, guid: 'some-guid' };
 
-      const _reportAbuse = sinon.stub().resolves(false);
+      mockAddonMananger
+        .expects('reportAbuse')
+        .withArgs(addon.guid)
+        .resolves(false);
 
-      _initiateAddonAbuseReportViaFirefox({ _reportAbuse, addon });
+      _initiateAddonAbuseReportViaFirefox({ addon });
 
       await sagaTester.waitFor(finishAddonAbuseReportViaFirefox().type);
-      sinon.assert.calledWith(_reportAbuse, addon.guid);
+      mockAddonMananger.verify();
     });
 
     it('dispatches loadAddonAbuseReport after a successful report', async () => {
       const addon = { ...fakeAddon, guid: 'some-guid' };
 
-      const _reportAbuse = sinon.stub().resolves(true);
+      mockAddonMananger.expects('reportAbuse').resolves(true);
 
-      _initiateAddonAbuseReportViaFirefox({ _reportAbuse, addon });
+      _initiateAddonAbuseReportViaFirefox({ addon });
 
       const expectedLoadAction = loadAddonAbuseReport({
         addon: { guid: addon.guid, id: addon.id, slug: addon.slug },
@@ -150,14 +155,16 @@ describe(__filename, () => {
 
       const loadAction = await sagaTester.waitFor(expectedLoadAction.type);
       expect(loadAction).toEqual(expectedLoadAction);
+
+      mockAddonMananger.verify();
     });
 
     it('does not dispatch loadAddonAbuseReport after a cancelled report', async () => {
       const addon = { ...fakeAddon, guid: 'some-guid' };
 
-      const _reportAbuse = sinon.stub().resolves(false);
+      mockAddonMananger.expects('reportAbuse').resolves(false);
 
-      _initiateAddonAbuseReportViaFirefox({ _reportAbuse, addon });
+      _initiateAddonAbuseReportViaFirefox({ addon });
 
       await sagaTester.waitFor(finishAddonAbuseReportViaFirefox().type);
 
@@ -167,6 +174,8 @@ describe(__filename, () => {
         reporter: null,
       });
       expect(sagaTester.numCalled(unexpectedAction.type)).toEqual(0);
+
+      mockAddonMananger.verify();
     });
   });
 });
```

Reply:
Duh. Why didn't I think of that? Thanks!